holidays/countries/trinidad_and_tobago.py

```diff
@@ -0,0 +1,160 @@
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays, _CustomHinduHolidays
+from holidays.calendars.gregorian import MAR, APR, MAY, JUN, JUL, AUG, OCT, NOV
+from holidays.groups import (
+    ChristianHolidays,
+    HinduCalendarHolidays,
+    InternationalHolidays,
+    IslamicHolidays,
+)
+from holidays.holiday_base import HolidayBase
+
+
+class TrinidadAndTobago(
+    HolidayBase, ChristianHolidays, HinduCalendarHolidays, InternationalHolidays, IslamicHolidays
+):
+    """
+    Trinidad and Tobago Holidays.
+
+    References:
+      * <https://en.wikipedia.org/wiki/Public_holidays_in_Trinidad_and_Tobago>
+      * <https://otp.tt/trinidad-and-tobago/national-holidays-and-awards/>
+      * <https://www.timeanddate.com/holidays/trinidad/eid-al-fitr>
+      * <https://calendarific.com/holiday/trinidad/eid-al-fitr>
+      * <https://www.timeanddate.com/holidays/trinidad/diwali>
+      * <https://calendarific.com/holiday/trinidad/diwali>
+      * <https://en.wikipedia.org/wiki/Trinidad_and_Tobago_Carnival>
+    """
+
+    country = "TT"
+    default_language = "en_TT"
+    # %s (estimated).
+    estimated_label = tr("%s (estimated)")
+    supported_languages = ("en_TT", "en_US")
+    # Trinidad and Tobago gained independence on August 31, 1962.
+    start_year = 1963
+
+    def __init__(self, islamic_show_estimated: bool = True, *args, **kwargs):
+        """
+        Args:
+            islamic_show_estimated:
+                Whether to add "estimated" label to Islamic holidays name
+                if holiday date is estimated.
+        """
+        ChristianHolidays.__init__(self)
+        HinduCalendarHolidays.__init__(self, cls=TrinidadAndTobagoHinduHolidays)
+        InternationalHolidays.__init__(self)
+        IslamicHolidays.__init__(
+            self, cls=TrinidadAndTobagoIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        self._add_new_years_day(tr("New Year's Day"))
+
+        if self._year >= 1996:
+            # Spiritual Baptist Shouter Liberation Day.
+            self._add_holiday_mar_30(tr("Spiritual Baptist Shouter Liberation Day"))
+
+            # Indian Arrival Day.
+            self._add_holiday_may_30(tr("Indian Arrival Day"))
+
+        # Labor Day.
+        self._add_holiday_jun_19(tr("Labour Day"))
+
+        # African Emancipation Day.
+        self._add_holiday_aug_1(tr("African Emancipation Day"))
+
+        # Independence Day.
+        self._add_holiday_aug_31(tr("Independence Day"))
+
+        # Republic Day.
+        self._add_holiday_sep_24(tr("Republic Day"))
+
+        # Diwali.
+        self._add_diwali(tr("Diwali"))
```

Comment:
```suggestion
        self._add_diwali(tr("Divali"))
```
Later we need to unify en_US name for this holiday (for now we can leave "Diwali", I think).

Reply:
okay!

Reply:
❓

Reply:
```suggestion
        dts_observed.update(self._add_diwali(tr("Divali")))
```